core/src/avm1/object/super_object.rs

```diff
@@ -18,44 +18,47 @@ use ruffle_macros::istr;
 /// with its parent class.
 #[derive(Copy, Clone, Collect)]
 #[collect(no_drop)]
-pub struct SuperObject<'gc>(Gc<'gc, SuperObjectData<'gc>>);
+pub struct SuperObject<'gc> {
+    /// The object present as `this` throughout the superchain.
+    this: Object<'gc>,
+
+    /// The prototype depth of the currently-executing method.
+    depth: u8,
+
+    /// Adds a niche, so that enums contaning this type can use it for their discriminant.
+    _niche: crate::utils::ZeroU8,
```

Comment:
How does this work exactly? Shouldn't Rust reuse this space for the discriminant anyway?

Reply:
Rust will never overlap a discriminant with padding, because padding can have *any* value (it's roughly like an invisible `MaybeUninit<[u8; _]>` field); adding the `ZeroU8` field instead says that any `SuperObject` *must* have a zero byte at this offset.